### PR TITLE
Implement unimplemented_method test

### PR DESCRIPTION
**Pull request by zsurocking**

### CODE REVIEW

**Comment by zsurocking** [comment]
#706

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 1]
```diff
@@ -811,6 +811,17 @@ public AccessToken refreshAccessToken() throws IOException {
...
   }
 
+  /** Sends an rpc to an unimplemented method on the server. **/
+  public void unimplementedMethod() {
+    UnimplementedServiceGrpc.UnimplementedServiceBlockingStub stub =
+        UnimplementedServiceGrpc.newBlockingStub(channel);
+    try {
+      stub.unimplementedCall(Empty.getDefaultInstance());
+    } catch (Throwable t) {
+      assertEquals(Status.UNIMPLEMENTED, Status.fromThrowable(t));
```
Comment (madongfly) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Per the [spec](https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#unimplemented_method), we should check the code and message.
```
assertEquals(Status.UNIMPLEMENTED.getCode(), Status.fromThrowable(t).getCode());
assertNull(Status.fromThrowable(t).getDescription());
```
I think the second check will fail with the current implementation, so we should either:
1. update the server side code to not append the description, or
2. remove this overstrict limit from the Spec and don't do such check.

@ejona86, what do you think?

Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: We could maybe do both. It does seem the test case is overly-strict. I know if we get a 404 from a server the client will say it came from an HTTP 404. We can talk to @a11r to determine why it requires the description to be null.

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 2]
```diff
@@ -811,6 +811,17 @@ public AccessToken refreshAccessToken() throws IOException {
         authScope.contains(response.getOauthScope()));
   }
 
+  /** Sends an rpc to an unimplemented method on the server. **/
+  public void unimplementedMethod() {
```
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Mark with `@Test`.

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 3]
```diff
@@ -811,6 +811,17 @@ public AccessToken refreshAccessToken() throws IOException {
...
+    UnimplementedServiceGrpc.UnimplementedServiceBlockingStub stub =
+        UnimplementedServiceGrpc.newBlockingStub(channel);
+    try {
+      stub.unimplementedCall(Empty.getDefaultInstance());
```
Comment (carl-mastrangelo) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: There needs to be a fail(); call after this line.

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 4]
```diff
@@ -811,6 +811,17 @@ public AccessToken refreshAccessToken() throws IOException {
...
+    } catch (Throwable t) {
```
Comment (carl-mastrangelo) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Catching Throwable will swallow the AssertionError that fail() throws.  Catch Exception instead.

Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Go ahead and make this `StatusRuntimeException`. That is the type we expect.

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java (hunk 3 above)
Comment (zsurocking) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: done

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java (hunk 2 above)
Comment (zsurocking) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: done

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java (hunk 1 above)
Comment (zsurocking) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Talked to @a11r, he would like to stick with no description if it does not add additional information.
Removed server side description.

**Comment by zsurocking** [comment]
PTAL

**Comment by madongfly** [comment]
@zsurocking, LGTM

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 5]
```diff
@@ -811,6 +812,22 @@ public AccessToken refreshAccessToken() throws IOException {
         authScope.contains(response.getOauthScope()));
   }
 
+  
+  /** Sends an rpc to an unimplemented method on the server. **/
+  @Test(timeout = 10000)
+  /** Sends an rpc to an unimplemented method on the server. **/
```
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Trash duplicate comment?

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 6]
```diff
@@ -811,6 +812,22 @@ public AccessToken refreshAccessToken() throws IOException {
         authScope.contains(response.getOauthScope()));
   }
 
+  
+  /** Sends an rpc to an unimplemented method on the server. **/
```
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: nit: Note that JavaDoc typically ends with just \*/, not \**/

**Comment by ejona86** [comment]
@zsurocking LGTM

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java (hunk 5 above)
Comment (zsurocking) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Done

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java (hunk 6 above)
Comment (zsurocking) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Done

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @zsurocking isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Pull request merged**